pkg/provisioning/service.go

```diff
@@ -170,39 +170,79 @@ func (s *Service) Delete(ctx context.Context, id string) error {
 // files that end with .yml or .yaml. It only reads files in the top-level
 // directory and skips non-regular files.
 func (s *Service) getYamlFiles(path string) ([]string, error) {
+	ctx := context.Background()
```

Comment:
We have a context in the caller, I'd just add it to the function parameters and get it from there.